source/grid/grid_generator.cc

```diff
@@ -8410,6 +8415,643 @@ namespace GridGenerator
+       {{1, 10, 2, 8, 7, 4, X, X, X, X, X, X, X, X, X, X}},
+       {{4, 9, 1, 4, 1, 7, 7, 1, 3, X, X, X, X, X, X, X}},
+       {{4, 9, 1, 4, 1, 7, 0, 8, 1, 8, 7, 1, X, X, X, X}},
+       {{4, 0, 3, 7, 4, 3, X, X, X, X, X, X, X, X, X, X}},
+       {{4, 8, 7, X, X, X, X, X, X, X, X, X, X, X, X, X}},
+       {{9, 10, 8, 10, 11, 8, X, X, X, X, X, X, X, X, X, X}},
+       {{3, 0, 9, 3, 9, 11, 11, 9, 10, X, X, X, X, X, X, X}},
+       {{0, 1, 10, 0, 10, 8, 8, 10, 11, X, X, X, X, X, X, X}},
+       {{3, 1, 10, 11, 3, 10, X, X, X, X, X, X, X, X, X, X}},
+       {{1, 2, 11, 1, 11, 9, 9, 11, 8, X, X, X, X, X, X, X}},
+       {{3, 0, 9, 3, 9, 11, 1, 2, 9, 2, 11, 9, X, X, X, X}},
+       {{0, 2, 11, 8, 0, 11, X, X, X, X, X, X, X, X, X, X}},
+       {{3, 2, 11, X, X, X, X, X, X, X, X, X, X, X, X, X}},
+       {{2, 3, 8, 2, 8, 10, 10, 8, 9, X, X, X, X, X, X, X}},
+       {{9, 10, 2, 0, 9, 2, X, X, X, X, X, X, X, X, X, X}},
+       {{2, 3, 8, 2, 8, 10, 0, 1, 8, 1, 10, 8, X, X, X, X}},
+       {{1, 10, 2, X, X, X, X, X, X, X, X, X, X, X, X, X}},
+       {{1, 3, 8, 9, 1, 8, X, X, X, X, X, X, X, X, X, X}},
+       {{0, 9, 1, X, X, X, X, X, X, X, X, X, X, X, X, X}},
+       {{0, 3, 8, X, X, X, X, X, X, X, X, X, X, X, X, X}},
+       {{X, X, X, X, X, X, X, X, X, X, X, X, X, X, X, X}}}};
+
+    static constexpr ndarray<unsigned int, n_lines, 2> line_to_vertex_table = {
+      {{{0, 1}},
+       {{1, 2}},
+       {{2, 3}},
+       {{3, 0}},
+       {{4, 5}},
+       {{5, 6}},
+       {{6, 7}},
+       {{7, 4}},
+       {{0, 4}},
+       {{1, 5}},
+       {{2, 6}},
+       {{3, 7}}}};
+
+    // run dimension-independent code
+    internal::process_sub_cell<3,
+                               n_vertices,
+                               n_sub_vertices,
+                               n_configurations,
+                               n_lines,
+                               16>(edgeTable,
+                                   triTable,
+                                   line_to_vertex_table,
+                                   ls_values,
+                                   points,
+                                   mask,
+                                   vertices,
+                                   cells);
+  }
+
+
+
+  template <int dim, typename VectorType>
+  void
+  create_triangulation_with_marching_cube_algorithm(
+    const MarchingCubeAlgorithm<dim, VectorType> &mc,
+    const DoFHandler<dim> &                       background_dof_handler,
+    const VectorType &                            ls_vector,
+    Triangulation<dim - 1, dim> &                 tria)
+  {
+    std::vector<Point<dim>>        vertices;
+    std::vector<CellData<dim - 1>> cells;
+    SubCellData                    subcelldata;
+
+    mc.process(background_dof_handler, ls_vector, vertices, cells);
+
+    std::vector<unsigned int> considered_vertices;
+
+    // GridTools::delete_duplicated_vertices (vertices, cells, subcelldata,
+    // considered_vertices);
```

Comment:
For some reasons, I have some issue when merging the triangles... But that is something for later.

Reply:
Isn't that a problem though? Let's say that two vertices that form an edge are duplicated. Don't you get two edges? What happens if you refine one cell but not the other? Does everything still works at the hanging node?